runtime/queries/metricsview_rows.go

```diff
@@ -100,21 +102,47 @@ func (q *MetricsViewRows) Resolve(ctx context.Context, rt *runtime.Runtime, inst
 	return nil
 }
 
-func (q *MetricsViewRows) Export(ctx context.Context, rt *runtime.Runtime, instanceID string, w io.Writer, opts *runtime.ExportOptions) error {
-	err := q.Resolve(ctx, rt, instanceID, opts.Priority)
+func (query *MetricsViewRows) generalExport(olap drivers.OLAPStore, mv *runtimev1.MetricsView, opts *runtime.ExportOptions, w io.Writer, rt *runtime.Runtime, instanceID string, ctx context.Context) error {
+	err := query.Resolve(ctx, rt, instanceID, opts.Priority)
 	if err != nil {
 		return err
 	}
 
-	mv, err := lookupMetricsView(ctx, rt, instanceID, q.MetricsViewName)
+	if opts.PreWriteHook != nil {
+		err = opts.PreWriteHook(query.generateFilename(mv))
+		if err != nil {
+			return err
+		}
+	}
+
+	switch opts.Format {
+	case runtimev1.ExportFormat_EXPORT_FORMAT_UNSPECIFIED:
+		return fmt.Errorf("unspecified format")
+	case runtimev1.ExportFormat_EXPORT_FORMAT_CSV:
+		return writeCSV(query.Result.Meta, query.Result.Data, w)
+	case runtimev1.ExportFormat_EXPORT_FORMAT_XLSX:
+		return writeXLSX(query.Result.Meta, query.Result.Data, w)
+	}
+
+	return nil
+}
+
+func duckDBCopyExport(sql string, args []any, filename string, olap drivers.OLAPStore, mv *runtimev1.MetricsView, opts *runtime.ExportOptions, w io.Writer, rt *runtime.Runtime, instanceID string, ctx context.Context) error {
```

Comment:
As the linter warning also indicates, can we move the args to `generalExport` and to `duckDBCopyExport` that are in common with `Export` to the start of the arg list for readability?

Reply:
Actually, since this function is also used for toplists, we should move it to a generic place similar to `writeCSV` funcs and others